### ISSUE TITLE
NCPA passive checks fail if allowed_hosts is set (haven't verified)

### ISSUE THREAD

**Issue by ericloyd** [NagiosEnterprises/ncpa#560]
We have a customer that reported the following after initial install and during their testing:  To summarize, they're trying to send passive checks via NRDP from the NCPA client to the XI server.  Their claim (I'm out of town and can't verify directly) is that if they have passive checks enabled, but enter hosts in allowed_hosts, then the passive checks fail with the error reproduced below.

Here's what they sent:
```
We ran into a problem with passive checks when restricting the IPs that are allowed to access the Web UI and API.

In the NCPA config, if we restrict the allowed_hosts, then the passive checks will not work.  The logs state that there is a permissions issue.  We have even tried allowing 127.0.0.1 and the actual IP of the agent node, but no dice.  See the error below.

If we remove the restriction, then the connection works.

So…is there a need to restrict the allowed hosts?  If, with the API, you can only read data, perhaps not.  However, if someone had the API key and could potentially modify the system, then we would want to restrict the hosts that can use the API (& Web GUI).
 

Sample Error from ncpa_passive log

2019-08-23 10:27:45,907:ERROR:ncpacheck:Error with JSON: No JSON object could be decoded. JSON was: <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>403 Forbidden</title>
<h1>Forbidden</h1>
<p>You don't have the permission to access the requested resource. It is either read-protected or not readable by the server.</p>
2019-08-23 10:27:45,907:ERROR:ncpa_windows:Stdout or returncode was None, cannot return meaningfully.
Traceback (most recent call last):
  File "C:\ncpa\agent\ncpa_windows.py", line 235, in run_all_handlers
  File "C:\ncpa\agent\passive\nrdp.py", line 115, in run
  File "C:\ncpa\agent\passive\nrdp.py", line 97, in get_xml_of_checkresults
  File "C:\ncpa\agent\passive\nrdp.py", line 56, in make_xml
  File "C:\ncpa\agent\passive\ncpacheck.py", line 84, in run
ValueError: Stdout or returncode was None, cannot return meaningfully.
```

**Comment by jomann09** [comment]
This sounds about right actually, I think that the passive checks actually use the internal API which likely does check the IP address. I will look into a fix for this. Also, for the question about needing to restrict the API. You cannot edit things using the API key, at least not in the current versions of NCPA. Unless of course you are using some plugin that does.

**Comment by jomann09** [comment]
This was fixed in 2.1.9 and should be out soon.
